diff --git a/Telegram/SourceFiles/logs.cpp b/Telegram/SourceFiles/logs.cpp
index e05553bbd006..550d335d9697 100644
--- a/Telegram/SourceFiles/logs.cpp
+++ b/Telegram/SourceFiles/logs.cpp
@@ -19,6 +19,18 @@ Full license: https://github.com/telegramdesktop/tdesktop/blob/master/LICENSE
 Copyright (c) 2014-2017 John Preston, https://desktop.telegram.org
 */
 #include "logs.h"
+#ifdef TDESKTOP_DISABLE_CRASH_REPORTS
+namespace SignalHandlers {
+	dump::~dump() {}
+	const dump &operator<<(const dump &stream, const char *str) { return stream; }
+	const dump &operator<<(const dump &stream, const wchar_t *str) { return stream; }
+	const dump &operator<<(const dump &stream, int num) { return stream; }
+	const dump &operator<<(const dump &stream, unsigned int num) { return stream; }
+	const dump &operator<<(const dump &stream, unsigned long num) { return stream; }
+	const dump &operator<<(const dump &stream, unsigned long long num) { return stream; }
+	const dump &operator<<(const dump &stream, double num) { return stream; }
+}
+#endif
 
 #include <signal.h>
 #include <new>
diff --git a/Telegram/gyp/Telegram.gyp b/Telegram/gyp/Telegram.gyp
index 89f7659d1117..20904336fbce 100644
--- a/Telegram/gyp/Telegram.gyp
+++ b/Telegram/gyp/Telegram.gyp
@@ -84,6 +84,7 @@
     'defines': [
       'AL_LIBTYPE_STATIC',
       '<!@(python -c "for s in \'<(build_defines)\'.split(\',\'): print(s)")',
+      'TDESKTOP_DISABLE_CRASH_REPORTS',
     ],
     'libraries': [
       # ls -d /usr/local/opt/*-td -w1|sed "s/^\(.*\)\@$/'-L\1\/lib',/"
diff --git a/Telegram/gyp/telegram_mac.gypi b/Telegram/gyp/telegram_mac.gypi
index 0eae831adc94..6882b4498ffd 100644
--- a/Telegram/gyp/telegram_mac.gypi
+++ b/Telegram/gyp/telegram_mac.gypi
@@ -64,9 +64,6 @@
       'PRODUCT_BUNDLE_IDENTIFIER': 'com.tdesktop.Telegram',
       'OTHER_CPLUSPLUSFLAGS': [ '-nostdinc++' ],
       'OTHER_LDFLAGS': [
-        '-lbase',
-        '-lcrashpad_client',
-        '-lcrashpad_util',
         '/usr/local/macold/lib/libz.a',
         '/usr/local/macold/lib/libopenal.a',
         '/usr/local/iconv_old/lib/libiconv.a',
@@ -119,13 +116,6 @@
       'action': [
         'mkdir', '-p', '${BUILT_PRODUCTS_DIR}/Telegram.app/Contents/Helpers/'
       ],
-    }, {
-      'postbuild_name': 'Copy crashpad_handler to Helpers',
-      'action': [
-        'cp',
-        '<(libs_loc)/crashpad_oldmac/crashpad/out/${CONFIGURATION}/crashpad_handler',
-        '${BUILT_PRODUCTS_DIR}/Telegram.app/Contents/Helpers/',
-      ],
     }],
   }, {
     'xcode_settings': {
@@ -164,9 +154,6 @@
     'xcode_settings': {
       'PRODUCT_BUNDLE_IDENTIFIER': 'com.tdesktop.Telegram',
       'OTHER_LDFLAGS': [
-        '-lbase',
-        '-lcrashpad_client',
-        '-lcrashpad_util',
       ],
      },
     'postbuilds': [{
@@ -186,13 +173,6 @@
       'action': [
         'mkdir', '-p', '${BUILT_PRODUCTS_DIR}/Telegram.app/Contents/Helpers/'
       ],
-    }, {
-      'postbuild_name': 'Copy crashpad_client to Helpers',
-      'action': [
-        'cp',
-        '<(libs_loc)/crashpad/crashpad/out/${CONFIGURATION}/crashpad_handler',
-        '${BUILT_PRODUCTS_DIR}/Telegram.app/Contents/Helpers/',
-      ],
     }],
   }], [ 'build_macstore', {
     'xcode_settings': {
